src/dependency/Dependency/ServiceProviderExtensions/Extensions.GetOrAbsent.cs

```diff
@@ -0,0 +1,32 @@
+#nullable enable
+
+using System;
+
+namespace PrimeFuncPack
+{
+    partial class ServiceProviderExtensions
+    {
+        public static Optional<T> GetServiceOrAbsent<T>(
+            this IServiceProvider serviceProvider)
+            where T : notnull
+            =>
+            InternalGetServiceOrAbsent<T>(
+                serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider)));
+
+        private static Optional<T> InternalGetServiceOrAbsent<T>(
+            IServiceProvider serviceProvider)
+            where T : notnull
+            =>
+            serviceProvider.GetService(typeof(T)) switch
+            {
+                T service
+                => Optional.Present(service),
```

Comment:
Style issue: let's change to Optional<T>.Present(service) to accord with Optional<T>.Absent (see below)

Reply:
Done